TripalCultivate/TripalCultivate
# G1.04 Content Type Collection:  Experiment Types

Issue #4 

## Description

This PR will create content types specific to Tripal cultivate. Currently it is focused on experiments types. I'm putting this up as a draft PR to let everyone know where I am and keep me accountable 🤪 

NOTE: The fields will not be ideal at this point. Often I will use a chado property for now but we will want drop downs, contact links, etc.

## Testing

Both the terms and the content types are imported on install of the module via a tripal job submission.
1. Create a fresh docker on this branch. This will submit two Tripal jobs and run them.
```
git clone https://github.com/TripalCultivate/TripalCultivate trpcult-contentTypes
cd trpcult-contentTypes
git checkout g1.04-contentTypes-Experiments
docker build --build-arg drupalversion=10.2.x-dev --build-arg phpversion=8.3 --tag=trpcultivate-base:Draft17 ./
docker run --publish=80:80 -tid --name=baseDraft17 --volume=`pwd`:/var/www/drupal/web/modules/contrib/TripalCultivate trpcultivate-base:Draft17
docker exec baseDraft17 service postgresql restart
```
2. Go to the Tripal Admin > Page Structure listing and confirm that the following experiment types are present:
![Screenshot 2024-04-10 at 5 04 26 PM](https://github.com/TripalCultivate/TripalCultivate/assets/1566301/a8bd4310-1c8d-43b2-bf04-9e7afd180223)
3. Use "Manage Fields" for each Experiment type and confirm that the fields are present as shown in screenshot.<img width="1341" alt="Screenshot 2024-04-11 at 11 23 19 AM" src="https://github.com/TripalCultivate/TripalCultivate/assets/1566301/1949b13a-e90e-45f1-98a3-d5e39b0e54b4">
4. Create experiments of each type and confirm that you can 🤪 and that the help text for each field makes sense.

**Example for Biochemical Experiment**
![image](https://github.com/TripalCultivate/TripalCultivate/assets/1566301/26dc330d-039f-4e0a-b689-b1c72f9212e1)
![image](https://github.com/TripalCultivate/TripalCultivate/assets/1566301/8b12e161-b4dc-42a0-bb82-5a41b9d2d6ec)

### Automated testing

I've also added an automated test which confirms that the content types are created and the expected number of fields are added. This keeps this module at 100% test coverage 😍
